src/dbnode/storage/entry.go

```diff
@@ -310,8 +310,19 @@ func (entry *Entry) TryMarkIndexGarbageCollected(reconciled, unreconciled tally.
 	if err != nil || e == nil {
 		return false
 	}
+
 	defer e.DecrementReaderWriterCount()
 
+	// Was reconciled if the entry retrieved from the shard differs from the current.
+	if e != entry {
+		reconciled.Inc(1)
+		entry.reverseIndex.RLock()
```

Comment:
Should we put the read locking within the Merge func?

Reply:
So the merge actually runs on the reconciled entry; we need to take a read lock here though since we're touching `entry.reverseIndex.states` on the entry itself to feed into the reconciled one

Reply:
Ah I see - good call